src/borg/archiver.py

```diff
@@ -1850,7 +1850,10 @@ def do_with_lock(self, args, repository):
         env = prepare_subprocess_env(system=True)
         try:
             # we exit with the return code we get from the subprocess
-            return subprocess.call([args.command] + args.args, env=env)
+            ret = subprocess.call([args.command] + args.args, env=env)
+        except (ValueError, FileNotFoundError, OSError) as e:
+            self.print_error(f"Could not execute '{args.command}': {e}")
```

Comment:
It is hard to tell whether this gives a reasonable error message for all the exceptions. We must try to not confuse users.

It is good that you included the original exception message into the error message, so that already helps.

"Could not execute" might be already slightly misleading though (might make people think of x-bit or permissions, but for FileNotFound, it is just not there).

Not sure what conditions lead to ValueError and misc. OSErrors here.

Maybe "Error while trying to run ..."?